pkg/cloud/cloud.go

```diff
@@ -279,6 +283,31 @@ func (c *cloud) CreateFileSystem(ctx context.Context, fileSystemId string, fileS
 	}, nil
 }
 
+func (c *cloud) ResizeFileSystem(ctx context.Context, fileSystemId string, newSizeGiB int64) (*int64, error) {
+	input := &fsx.UpdateFileSystemInput{
+		FileSystemId:    aws.String(fileSystemId),
+		StorageCapacity: aws.Int64(newSizeGiB),
+	}
+
+	_, err := c.fsx.UpdateFileSystemWithContext(ctx, input)
+	if err != nil {
+		if awsErr, ok := err.(awserr.Error); ok {
+			if awsErr.Code() == fsx.ErrCodeBadRequest &&
+				awsErr.Message() == "Unable to perform the storage capacity update. There is an update already in progress." {
+				// If a previous request timed out and was successful, don't error
+				_, err = c.getUpdateResizeFilesystemAdministrativeAction(ctx, fileSystemId, newSizeGiB)
+				if err != nil {
+					return nil, err
+				}
+			}
+		} else {
+			return nil, fmt.Errorf("UpdateFileSystem failed: %v", err)
```

Comment:
Yea, since Lustre's method returns a primitive instead of a pointer they need to return a value. Seems like they opted to return the original size instead of 0 but it is never used.